src/ast/nodes/MemberExpression.ts

```diff
@@ -95,19 +96,49 @@ export default class MemberExpression
 	extends NodeBase
 	implements DeoptimizableEntity, ChainElement
 {
-	declare computed: boolean;
 	declare object: ExpressionNode | Super;
-	declare optional: boolean;
 	declare property: ExpressionNode | PrivateIdentifier;
-	declare propertyKey: ObjectPathKey | null;
+	declare propertyKey?: ObjectPathKey;
 	declare type: NodeType.tMemberExpression;
-	variable: Variable | null = null;
+	variable?: Variable;
 	protected declare assignmentInteraction: NodeInteractionAssigned;
 	private declare accessInteraction: NodeInteractionAccessed;
-	private assignmentDeoptimized = false;
-	private bound = false;
 	private expressionsToBeDeoptimized: DeoptimizableEntity[] = [];
-	private isUndefined = false;
+
+	get computed(): boolean {
+		return isFlagSet(this.flags, Flag.computed);
+	}
+	set computed(value: boolean) {
+		this.flags = setFlag(this.flags, Flag.computed, value);
+	}
+
+	get optional(): boolean {
+		return isFlagSet(this.flags, Flag.optional);
+	}
+	set optional(value: boolean) {
+		this.flags = setFlag(this.flags, Flag.optional, value);
+	}
+
+	private get assignmentDeoptimized(): boolean {
+		return isFlagSet(this.flags, Flag.assignmentDeoptimized);
+	}
+	private set assignmentDeoptimized(value: boolean) {
+		this.flags = setFlag(this.flags, Flag.assignmentDeoptimized, value);
+	}
+
+	private get bound(): boolean {
+		return isFlagSet(this.flags, Flag.bound);
+	}
+	private set bound(value: boolean) {
+		this.flags = setFlag(this.flags, Flag.bound, value);
+	}
+
+	private get isUndefined(): boolean {
+		return isFlagSet(this.flags, Flag.isUndefined);
+	}
+	private set isUndefined(value: boolean) {
+		this.flags = setFlag(this.flags, Flag.isUndefined, value);
+	}
 
```

Comment:
Yes, that would be even better.

Reply:
I agree that this might still be somewhat beneficial, but for the few instances that I looked into, there were a lot of code changes needed to get there. I would prefer to do that work in a separate commit if possible, just to try to reduce the scope of this PR. @lukastaegert Feel free to unresolve this conversation if you would still like to see this done as part of this PR instead.